src/lib/components/Textfield.react.js

```diff
@@ -0,0 +1,139 @@
+    multiline?: string,
+    fullWidth?: string,
+    input?: string,
+    inputMultiline?: string,
+  },
+  /** CSS class name of the root element */
+  className?: string,
+  /** 'primary' | 'secondary' */
+  color: string,
+  /** If true, the input element will be disabled. */
+  disabled?: boolean,
+  /** If true, the label will be displayed in an error state */
+  error?: boolean,
+  /** Dash callback to trigger an event handler */
+  fireEvent?: () => void,
+  /** If true, the input will take up the full width of its container */
+  fullWidth?: boolean,
+  /** The helper text content */
+  helperText?: Node,
+  /** Element ID */
+  id?: string,
+  /** If true, a textarea element will be rendered instead of an input */
+  multiline?: boolean,
+  /** If true, the label is displayed as required and the input element` will be required */
+  required?: boolean,
+  /** Dash callback to update props on the server. */
+  setProps?: () => void,
+  /** 'medium' | 'small' */
+  size?: string,
+  /** Override the inline-styles of the root element */
+  style?: Object,
+  /** Type of the input element. It should be a valid HTML5 input type */
+  type?: string,
+  /** The value of the input element, required for a controlled component */
+  value?: any,
+  /** 'filled' | 'outlined' | 'standard' */
+  variant?: string,
+};
+
+type State = {
+  disabled: boolean,
+  value: any,
+};
+
+const defaultProps = {
+  autoFocus: false,
+  classes: {},
+  className: '',
+  color: 'primary',
+  disabled: false,
+  error: false,
+  fireEvent: () => {},
+  fullWidth: false,
+  helperText: '',
+  multiline: false,
+  required: false,
+  setProps: () => {},
+  size: 'medium',
+  style: {},
+  type: 'text',
+  value: '',
+  variant: 'standard'
+};
+
+/** Material UI TextField component */
+export default class TextField extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      disabled: this.props.disabled,
+      value: this.props.value,
+    };
+    /** _.debounce used to provide delay in callback to avoid firing callback every
+     * time user input changes - waits this.props.dashCallbackDelay ms to fire callback */
+    this.updateTextProps = _.debounce(this._updateTextProps, this.props.dashCallbackDelay);
+  }
+
+  handleChange = (event) => {
+    this.setState({value: event.target.value});
+
+    if (typeof this.props.setProps === 'function') {
+      this.props.setProps({value: event.target.value});
+    }
+  };
+
+  render() {
+    const { id, className, style} = this.props;
+    this.handleChange = this.handleChange.bind(this)
+    this.filterFunc = this.filterFunc.bind(this)
+    this.updateTextProps = this.updateTextProps.bind(this)
+    return (
+      <div className={className}>
+        <MuiThemeProvider muiTheme={getMuiTheme(lightBaseTheme)}>
+          <MuiTextField
+            id={id}
+            className={className}
+            style={style}
+          />
+        </MuiThemeProvider>
```

Comment:
There are some props which you defined above but did not pass into this component, such as `disabled`, `value` and a few others. Would you include all of them here?